x-pack/plugins/ml/public/application/components/data_grid/column_chart.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+
+import { BarSeries, Chart, Settings } from '@elastic/charts';
+import { EuiDataGridColumn } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import './column_chart.scss';
+
+import {
+  isNumericChartData,
+  isOrdinalChartData,
+  useColumnChart,
+  ChartData,
+} from './use_column_chart';
+
+const getLegendText = (chartData: ChartData, MAX_CHART_COLUMNS: number): string | JSX.Element => {
+  if (chartData.type === 'boolean') {
+    return (
+      <table className="mlDataGridChart__legendBoolean">
+        <tbody>
+          <tr>
+            <td>{chartData.data[0].key_as_string}</td>
+            <td>{chartData.data[1].key_as_string}</td>
+          </tr>
+        </tbody>
+      </table>
+    );
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality === 1) {
+    return i18n.translate('xpack.ml.dataGridChart.singleCategoryLegend', {
+      defaultMessage: `{cardinality} category`,
+      values: { cardinality: chartData.cardinality },
+    });
+  }
+
+  if (isOrdinalChartData(chartData) && chartData.cardinality > MAX_CHART_COLUMNS) {
+    return i18n.translate('xpack.ml.dataGridChart.topCategoriesLegend', {
+      defaultMessage: `top {MAX_CHART_COLUMNS} of {cardinality} categories`,
+      values: { cardinality: chartData.cardinality, MAX_CHART_COLUMNS },
+    });
+  }
+
+  if (
+    isOrdinalChartData(chartData) &&
+    chartData.cardinality <= MAX_CHART_COLUMNS &&
+    chartData.cardinality > 1
+  ) {
+    return i18n.translate('xpack.ml.dataGridChart.categoriesLegend', {
+      defaultMessage: `{cardinality} categories`,
```

Comment:
can be combined with `singleCategoryLegend ` using pluralize

Reply:
Fixed in 4a66f7d.